### PR TITLE
gcp - add kubernetes initial setup

### PR DESCRIPTION
**Pull request by Dkhanas**
Added initial classes with query and get  for projects locations resource in Kubernetes service

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/kubernetes.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo
+
+
+@resources.register('k8s-project-location')
```
Comment (bclements) [inline tools/c7n_gcp/c7n_gcp/resources/kubernetes.py]: Change k8s to kubernetes

Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/kubernetes.py]: Done

File: tools/c7n_gcp/c7n_gcp/resources/kubernetes.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Copyright 2018-2019 Capital One Services, LLC
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/kubernetes.py]: just 2019

File: tools/c7n_gcp/tests/tests_kubernates.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Copyright 2018 Capital One Services, LLC
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/tests_kubernates.py]: should be 2019

File: tools/c7n_gcp/c7n_gcp/resources/kubernetes.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+class KubernatesProjectLocation(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'container'
+        version = 'v1'
+        component = 'projects.locations.clusters'
+        enum_spec = ('list', 'clusters[]', None)
+        scope = 'project'
+        scope_key = 'parent'
+        scope_template = 'projects/{}/locations/-'
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/kubernetes.py]: is this `-` in `scope_template` intentional? I'm just unfamiliar with the syntax for that in this context

Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/kubernetes.py]: "-" means we should get data from all locations. More information you can get from [Kubernetes Engine Method: projects.locations.clusters.list](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters/list)

File: tools/c7n_gcp/tests/tests_kubernates.py (hunk 3 above)
Comment (Dkhanas) [reply tools/c7n_gcp/tests/tests_kubernates.py]: Done

File: tools/c7n_gcp/c7n_gcp/resources/kubernetes.py (hunk 2 above)
Comment (Dkhanas) [reply tools/c7n_gcp/c7n_gcp/resources/kubernetes.py]: Done

**Comment by egor-miasnikov** [comment]
@bclements @jondavidi is this PR ready for going to `upstream`? If so please click `Approve`

File: tools/c7n_gcp/c7n_gcp/resources/kubernetes.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo
+
+
+@resources.register('kubernetes-project-location')
+class KubernatesProjectLocationCluster(QueryResourceManager):
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/kubernetes.py]: still a few more changes needed specifically to change anywhere that has `kubernetes` in a string or class name to be `container` to match the actual service name. This includes the file name. 

Please reference comments on this PR: https://github.com/mediapills/cloud-custodian/pull/19

File: tools/c7n_gcp/c7n_gcp/entry.py [hunk 6]
```diff
@@ -25,6 +25,7 @@
 import c7n_gcp.resources.iam
 import c7n_gcp.resources.logging
 import c7n_gcp.resources.network
+import c7n_gcp.resources.kubernetes
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/entry.py]: change to `container`. see other comments

File: tools/c7n_gcp/c7n_gcp/resources/kubernetes.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo
+
+
+@resources.register('kubernetes-project-location')
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/kubernetes.py]: change to `container`

**Comment by mediapills** [comment]
@jondavidi All changes applied.

**Comment by Dkhanas** [comment]
This PR was closed in reason of investigation. We found Kubernates cluster was implemented in gke.py file. So i moved all classes into that file. New PR is here https://github.com/mediapills/cloud-custodian/pull/45
class names i left as it was in that file, because cluster already pushed in upstream. So i believe we should be similar with upstream resource names.

**Pull request closed without merging**